src/main/java/bio/terra/folder/db/FolderDao.java

```diff
@@ -0,0 +1,102 @@
+package bio.terra.folder.db;
+
+import bio.terra.folder.db.exception.FolderNotFoundException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.openapitools.jackson.nullable.JsonNullable;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class FolderDao {
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  @Autowired
+  public FolderDao(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  public void createFolder(
+      String folderId,
+      String folderName,
+      JsonNullable<String> parentFolderId,
+      JsonNullable<String> spendProfile,
+      boolean spendProfileInherited) {
+    String sql =
+        "INSERT INTO folder (folder_id, folder_name, parent_folder_id, spend_profile_id, spend_profile_inherited)"
+            + "values (:folder_id, :folder_name, :parent_folder_id, :spend_profile_id, :spend_profile_inherited)";
+    Map<String, Object> paramMap = new HashMap<>();
+    paramMap.put("folder_id", folderId);
+    paramMap.put("folder_name", folderName);
+    paramMap.put("spend_profile_inherited", spendProfileInherited);
+
+    if (parentFolderId.isPresent() && parentFolderId.get() != null) {
+      paramMap.put("parent_folder_id", parentFolderId.get());
+    } else {
+      paramMap.put("parent_folder_id", null);
+    }
+
+    if (spendProfile.isPresent() && spendProfile.get() != null) {
+      paramMap.put("spend_profile_id", spendProfile.get());
+    } else {
+      paramMap.put("spend_profile_id", null);
+    }
+
+    jdbcTemplate.update(sql, paramMap);
+  }
+
+  public boolean deleteFolder(String folderId) {
+    String sql = "DELETE FROM folder WHERE folder_id = :id";
```

Comment:
how are we handling deleting folders that have children? is it just going to throw a FK exception or are we going to cascade? "this is only used by undoStep for now so it hasn't been considered" is also a reasonable answer- i'm more thinking out loud here

Reply:
Right now, my thinking is "this is only used by undoStep, so as long as transactional properties hold that won't be an issue here".
There's some thoughts about a delete endpoint in the design - at the moment we're leaning towards only letting users delete empty folders, and possibly providing a non-atomic delete for non-empty folders.